blivet/udev.py

```diff
@@ -204,7 +204,11 @@ def device_get_name(udev_info):
         name = udev_info["DM_NAME"]
     elif "MD_DEVNAME" in udev_info:
         mdname = udev_info["MD_DEVNAME"]
-        if device_is_partition(udev_info):
+        if device_is_disk(udev_info):
+            # this device is actually not an MD device but a disk from the array
+            # these sometimes have the MD_* properties in udev database
+            name = udev_info["SYS_NAME"]
+        elif device_is_partition(udev_info):
```

Comment:
See https://github.com/storaged-project/blivet/pull/785/commits/b4e9cee5f0490c84bed4a390ff27fea0f38e4462 for an alternative solution to this which covers partitions as well.

Reply:
Interesting. This fixes the crash, the array is added correctly to the tree, but not the partitions on it. In my case the name of the array is actually `127`, so we need to try to add partitions with name `127p1 because `md127p1` doesn't work with parted:

```
DEBUG:blivet: looking up parted Partition: /dev/md/md127p1
ERROR:blivet: Failed to instantiate PartitionDevice: ('cannot find parted partition instance', 'md127p1')
DEBUG:blivet: no device obtained for md127p1
```

But from the bug mentioned in your patch it looks like the problem there was completely opposite -- we were trying to add partition `/dev/10p3` instead of `/dev/md/10p3` or `/dev/md10p3`.